ktor-server/ktor-server-plugins/ktor-server-call-logging/jvm/test/io/ktor/server/plugins/callloging/CallLoggingTest.kt

```diff
@@ -531,6 +532,33 @@ class CallLoggingTest {
         }
     }
 
+    @Test
+    fun `dont override MDC with configured entry`() = testApplication {
+        environment {
+            log = logger
+        }
+        application {
+            install(CallLogging) {
+                mdc("hardcoded") { "1" }
+                mdc("mdc-uri") { it.request.uri }
+
+                format { it.request.uri }
+            }
+        }
+        routing {
+            get {
+                MDC.put("name", "<name>")
+
+                withContext(MDCContext()) {
+                    call.respondText { "OK" }
+                }
+            }
+        }
+
+        client.get("/")
+        assertContains(messages, "INFO: / [hardcoded=1, mdc-uri=/, name=<name>]")
```

Comment:
I think we could use another assertion to ensure that the MDC is the reverted to the original state after the context of the call.  I don't know if this is how it works now, but it's how I imagine it _should_ work 🤔 

So something like:
```kotlin
client.get("/")
logger.info("call from outside ktor")

assertContains(messages, "INFO: / [hardcoded=1, mdc-uri=/, name=<name>]")
assertTrue { messages.any { "name" in it && "mdc-uri" !in it } }
```

Reply:
Yes, thank you a lot, I will change